File: src/01-easy/power-of-thor-episode-1/solution.dart
import 'dart:io';
void main() {
    var s = stdin.readLineSync()!.split(' ');
    int lx = int.parse(s[0]), ly = int.parse(s[1]);
    int tx = int.parse(s[2]), ty = int.parse(s[3]);
    while (true) {
        stdin.readLineSync();
        var dir = '';
        if (ty > ly) { dir += 'N'; ty--; }
        else if (ty < ly) { dir += 'S'; ty++; }
        if (tx > lx) { dir += 'W'; tx--; }
        else if (tx < lx) { dir += 'E'; tx++; }
        print(dir);
    }
}

File: src/01-easy/power-of-thor-episode-1/solution.rs
use std::io;
fn main() {
    let mut s = String::new();
    io::stdin().read_line(&mut s).unwrap();
    let v: Vec<i32> = s.trim().split(' ').map(|x| x.parse().unwrap()).collect();
    let (lx, ly) = (v[0], v[1]);
    let (mut tx, mut ty) = (v[2], v[3]);
    loop {
        let mut buf = String::new();
        io::stdin().read_line(&mut buf).unwrap();
        let mut dir = String::new();
        if ty > ly { dir.push('N'); ty -= 1; }
        else if ty < ly { dir.push('S'); ty += 1; }
        if tx > lx { dir.push('W'); tx -= 1; }
        else if tx < lx { dir.push('E'); tx += 1; }
        println!("{}", dir);
    }
}
